tests/Bot.Tests/Common/Recognizer/TimeSurveyBotLuisExTest.cs

```diff
@@ -117,9 +120,351 @@ public void TimePeriodInMinutes_EightHoursPeriod_ReturnsMinutes()
             };
 
             const double expectedMinutes = 480.00;
-            
+
             luisResult.WorkedDurationInMinutes().Should().Be(expectedMinutes);
         }
-        
+
+        [Fact]
+        public void WorkedPeriod_DateWithoutTime_ReturnsWorkedPeriodFromNineAm()
+        {
+            var mondayFirstNovember = new DateTime(2021, 11, 8, 15, 0, 0);
+            var lastFridayStart = new DateTime(2021, 11, 5, 6, 0, 0);
+            var lastFridayEnd = new DateTime(2021, 11, 5, 8, 0, 0);
+            var timeZonePlusThree = TZConvert.GetTimeZoneInfo("Europe/Moscow");
+
+            var mockDateTimeProvider = new Mock<IDateTimeProvider>();
+            mockDateTimeProvider.Setup(d => d.DateTimeUtcNow())
+                .Returns(mondayFirstNovember);
+
+            var instances = new TimeSurveyBotLuis._Entities._Instance
+            {
+                datetime = new[]
+                {
+                    new InstanceData
+                    {
+                        Text = "2 hours",
+                        Type = "builtin.datetimeV2.duration"
+                    },
+                    new InstanceData
+                    {
+                        Text = "last friday",
+                        Type = "builtin.datetimeV2.datetime"
+                    }
+                }
+            };
+
+            var luisResult = new TimeSurveyBotLuis
+            {
+                Entities = new TimeSurveyBotLuis._Entities
+                {
+                    _instance = instances
+                }
+            };
+
+            var (start, end) = luisResult.WorkedPeriod(mockDateTimeProvider.Object, 120,
+                timeZonePlusThree);
+
+            start.Should().Be(lastFridayStart);
+            end.Should().Be(lastFridayEnd);
+        }
+
+        [Fact]
+        public void WorkedPeriod_PeriodStartingFromDateTime_ReturnsWorkedPeriod()
+        {
+            var mondayFirstNovember = new DateTime(2021, 11, 8, 15, 0, 0);
```

Comment:
I don't understand the name of this variable

Reply:
You're right, I change the datetime without updating the variable's name